usr/lib/ep11_stdll/ep11_specific.c

```diff
@@ -5546,6 +6249,23 @@ CK_RV ep11tok_sign_init(STDLL_TokData_t * tokdata, SESSION * session,
         goto done;
     }
 
+    rc = ep11tok_pkey_check(tokdata, session, key_obj, mech);
+    switch (rc) {
+    case CKR_OK:
+        rc = sign_mgr_init(tokdata, session, ctx, mech, recover_mode, key);
+        if (rc == CKR_OK) {
+            ctx->pkey_active = TRUE;
+            free(ep11_sign_state);
+            goto done;
+        }
+        break;
```

Comment:
So if ep11tok_pkey_check() return d CKR_OK, but sign_mgr_init failed, the  you now go to the EP11 path? Is this intended? 
ep11tok_pkey_check() might have unlocked amd relocked the object, so keyblob obtained from the object before the object was unlocked, so it may no longer be valid. 

I guess if sign_mgr_init() failes here, you should always go to out. (and always free(ep11_sign_state) as well).

Reply:
Done.

Reply:
Done.